### ISSUE TITLE
Add support for filtering based on Kafka message headers

### ISSUE THREAD

**Issue by danphanley** [confluentinc/ksql#658]
From the source topic, do filtering based on header of the message, do no modification of the body, and write to a destination topic

**Comment by joewood** [comment]
Is this still on the roadmap?

**Comment by apurvam** [comment]
This is a great feature request, but it isn't on the roadmap at present.

**Comment by joewood** [comment]
Would this be that difficult given that KIP-244 has landed in Kafka 2?

**Comment by apurvam** [comment]
Yes, since that KIP only exposes headers in the processor API, not the DSL, which is what KSQL uses exclusively right now. Beyond the technical challenges, we also need to work out the language extensions for exposing the header information.

**Comment by joewood** [comment]
Oh, didn't realize that. We really need Streams DSL to support headers too. :-) Surprised there's not an active dedicated KIP for that.

**Comment by Georgi-Ivanov-Fourth** [comment]
Filtering based on message headers is must-have for multi-tenant scenarios. Otherwise the tenant meta-data should become part of the message payload - not a deal breaker, but there should be a better way.

**Comment by big-andy-coates** [comment]
Closing as a duplicate of https://github.com/confluentinc/ksql/issues/5167

**Comment by joewood** [comment]
OK, np. Can we keep the original issues open, that saves re-subscribing to new issues covering a 2 year old request?
